src/pages/Account/Profile.tsx

```diff
@@ -44,8 +45,10 @@ const Profile: React.FunctionComponent<PropsType> = ({ location }) => {
   const onSubmit = async (data: ProfileFormData) => {
     if (!user) return;
     await updateUserProfile(user.uid, data);
-    const accountQuestionsUrl = `/account/questions?venueId=${venueId}&returnUrl=${window.location.pathname}${window.location.search}`;
-    const nextUrl = venueId ? accountQuestionsUrl : `/${DEFAULT_VENUE}`;
+    const accountQuestionsUrl = `/account/questions?venueId=${venueId}${
+      returnUrl ? "&returnUrl=" + returnUrl : ""
+    }`;
+    const nextUrl = venueId ? accountQuestionsUrl : returnUrl?.toString() ?? "";
     history.push(IS_BURN ? `/enter/step3` : nextUrl);
```

Comment:
[Future] We should be stripping out all of this `IS_BURN` logic from this flow as well.